aptos-move/e2e-move-tests/tests/rotate_auth_key.rs

```diff
@@ -63,6 +109,88 @@ fn rotate_auth_key_twice() {
+    new_public_key: MultiEd25519PublicKey,
+) {
+    // construct a proof challenge struct that proves that
+    // the user intends to rotate their auth key
+    let rotation_proof = RotationProofChallenge {
+        account_address: CORE_CODE_ADDRESS,
+        module_name: String::from("account"),
+        struct_name: String::from("RotationProofChallenge"),
+        sequence_number,
+        originator,
+        current_auth_key: AccountAddress::from_bytes(&current_account.auth_key()).unwrap(),
+        new_public_key: new_public_key.to_bytes().to_vec(),
+    };
+
+    let rotation_msg = bcs::to_bytes(&rotation_proof);
+
+    // sign the rotation message by the current private key and the new private key
+    let signature_by_curr_privkey = current_account
+        .privkey
+        .sign_arbitrary_message(&rotation_msg.clone().unwrap());
+    let signature_by_new_privkey = new_private_key.sign_arbitrary_message(&rotation_msg.unwrap());
+
+    assert_success!(harness.run_transaction_payload(
+        &current_account,
+        aptos_stdlib::account_rotate_authentication_key(
+            0,
+            current_account.pubkey.to_bytes().to_vec(),
+            1,
+            new_public_key.to_bytes().to_vec(),
+            signature_by_curr_privkey.to_bytes().to_vec(),
+            signature_by_new_privkey.to_bytes().to_vec(),
+        )
+    ));
+}
+
+pub fn assert_successful_payload_key_rotation_from_ed25519_to_ed25519(
+    harness: &mut MoveHarness,
+    current_account: Account,
+    originator: AccountAddress,
+    sequence_number: u64,
+    new_private_key: Ed25519PrivateKey,
+    new_public_key: Ed25519PublicKey,
+) {
+    // construct a proof challenge struct that proves that
+    // the user intends to rotate their auth key
+    let rotation_proof = RotationProofChallenge {
+        account_address: CORE_CODE_ADDRESS,
+        module_name: String::from("account"),
+        struct_name: String::from("RotationProofChallenge"),
+        sequence_number,
+        originator,
+        current_auth_key: AccountAddress::from_bytes(&current_account.auth_key()).unwrap(),
+        new_public_key: new_public_key.to_bytes().to_vec(),
+    };
+
+    let rotation_msg = bcs::to_bytes(&rotation_proof);
+
+    // sign the rotation message by the current private key and the new private key
+    let signature_by_curr_privkey = current_account
+        .privkey
+        .sign_arbitrary_message(&rotation_msg.clone().unwrap());
+    let signature_by_new_privkey = new_private_key.sign_arbitrary_message(&rotation_msg.unwrap());
+
+    assert_success!(harness.run_transaction_payload(
+        &current_account,
+        aptos_stdlib::account_rotate_authentication_key(
+            0,
+            current_account.pubkey.to_bytes().to_vec(),
+            0,
+            new_public_key.to_bytes().to_vec(),
+            signature_by_curr_privkey.to_bytes().to_vec(),
+            signature_by_new_privkey.to_bytes().to_vec(),
+        )
+    ));
+}
+
```

Comment:
Again, perhaps in the next PR, we should also test the following cases:
1. Rotation from MultiEd25519 to Ed25519:
2. Rotation from MultiEd25519 to MultiEd25519: 

Making your functions generic using the `aptos-crypto` traits will help with this.

Reply:
good point - will do that in the next pr : )

Reply:
Cool! We'll need to make the `Account` struct generic over any signature scheme [here](https://github.com/aptos-labs/aptos-core/blob/04e39cb28e1b3aef281fbc6993d090d2f7d838e9/aptos-move/e2e-tests/src/account.rs#L37)